.github/workflows/tox-pytest.yml

```diff
@@ -49,49 +139,78 @@ jobs:
         uses: actions/cache@v3
         id: cache-zenodo-datastore
         with:
-          path: ~/pudl-work/data/
+          path: ${{ env.PUDL_INPUT }}
           key: zenodo-datastore-${{ hashFiles('datastore-dois.txt') }}
-          restore-keys: |
-            zenodo-datastore-
 
-      - name: Log pre-test Zenodo datastore contents
-        run: |
-          mkdir -p ~/pudl-work/data/
-          find ~/pudl-work/data/
+      - name: Make cache/output dirs
+        run: mkdir -p ${{ env.PUDL_OUTPUT }} ${{ env.PUDL_INPUT}}
 
-      - name: Log SQLite3 version
-        run: |
-          which sqlite3
-          sqlite3 --version
+      - name: List workspace contents
+        run: find ~/pudl-work
 
-      - name: Set default gcp credentials
+      - name: Set default GCP credentials
         id: gcloud-auth
         continue-on-error: true
         uses: "google-github-actions/auth@v1"
         with:
           workload_identity_provider: "projects/345950277072/locations/global/workloadIdentityPools/gh-actions-pool/providers/gh-actions-provider"
           service_account: "tox-pytest-github-action@catalyst-cooperative-pudl.iam.gserviceaccount.com"
 
-      - name: Run PyTest with Tox against GCS cache (internal PRs)
+      - name: Run integration tests, trying to use GCS cache if possible
         run: |
-          tox -- --gcs-cache-path gs://zenodo-cache.catalyst.coop
-        if: ${{ steps.gcloud-auth.outcome == 'success' }}
+          tox -e integration -- --gcs-cache-path=gs://zenodo-cache.catalyst.coop --durations 0
 
-      - name: Run PyTest with Tox against Zenodo (external PRs)
+      - name: Rename coverage file
         run: |
-          tox
-        if: ${{ steps.gcloud-auth.outcome == 'failure' }}
+          mv coverage.xml coverage-integration.xml
 
+      - name: Upload coverage
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-integration
+          path: coverage-integration.xml
       - name: Log post-test Zenodo datastore contents
-        run: find ~/pudl-work/data/
+        run: find ${{ env.PUDL_INPUT }}
 
+  ci-coverage:
+    runs-on: ubuntu-latest
+    needs:
+      - ci-unit
+      - ci-integration
```

Comment:
Does this need to include `ci-static` as well if that's the only place where coverage is being generated from the docs build?

Reply:
Oh we're not trying to get that coverage working yet.

What kind of flaky/buggy is it right now? Is it just the upload to CodeCov? I haven't done anything beyond the default setup there. Not sure how uploading a directory of coverage files work. Maybe there's some local concatenation that we could/should do before upload?

Reply:
The issue with Codecov was that I wasn't checking out the code in the upload job before trying to upload coverage reports to codecov. Which meant that the coverage reports were being uploaded without proper path mapping..